src/http.rs

```diff
@@ -135,6 +142,17 @@ where
 
         Box::pin(fut)
     }
+
+    fn maybe_add_proxy_headers(&self, request: &mut Request<B>) {
+        if let Some(proxy_headers) = self.proxy_connector.http_headers(request.uri()) {
+            for (k, v) in proxy_headers {
+                let request_headers = request.headers_mut();
+                if !request_headers.contains_key(k) {
```

Comment:
Do we need to check if it exists, or can we just insert the new header? HTTP headers can contain multiple values.

Reply:
I'm going by RFC 7235, https://github.com/hyperium/headers/blob/f01cc90cf8d601a716856bc9d29f47df92b779e4/src/common/authorization.rs#L9 and https://github.com/hyperium/headers/blob/master/src/common/proxy_authorization.rs#L3. My read of RFC 7235 is that it doesn't leave room for >1 sets of credentials in the Authorization and Proxy-Authorization headers.
The reason I added the existence check - https://github.com/vectordotdev/vector/blob/53e178570b5b87bc2124f4299865cbb00916fe20/src/http.rs#L293 can also set the Authorization header, which I didn't want to overwrite.